modules/avatar/react/lib/Avatar.tsx

```diff
@@ -30,60 +31,132 @@ export interface AvatarLocalProps {
    * The url of the Avatar image.
    */
   url?: string;
+  /**
+   * The alternative container type for the button. Uses Emotion's special `as` prop.
+   * Will render an `div` tag instead of a `button` when defined.
+   */
+  as?: 'div';
+  /**
+   * The function called when the Avatar is clicked.
+   */
+  onClick?: React.MouseEventHandler<HTMLButtonElement>;
+  /**
+   * The ref to the button or div that the styled component renders.
+   */
+  ref?: React.Ref<HTMLButtonElement>;
```

Comment:
I don't think this will work. Let me test it out.

Reply:
I considered adding the ref in the Avatar overload but wasn't sure if it was needed since in Alex's example it wasn't split up.

Reply:
I have confirmed this will not work and React will throw the following warning:
```
react.development.js:167 Warning: Avatar: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop.
```

I think `ref` is the correct prop, but we'll have to use `forwardRef` for the component to make that work. Some more details here: #548

Reply:
Good catch 👍  We should make sure we have a test for this as well.